### DISCUSSION TITLE
Feature request: AlgoliaSearch 5 support

### DISCUSSION THREAD

**Discussion by leonard-henriquez** [algolia/instantsearch#5881] [Ideas]
Do you have an ETA on when InstantSearch will be compatible with the AlgoliaSearch 5 client ?

**Comment by Haroenv** [comment]
@leonard-henriquez, could you clarify how it isn't supported? The types should support it, and the code definitely does

**Reply by aliou** [reply]
Hey @Haroenv, I'm a teammate of @leonard-henriquez and taking over this thread. Using the AlgoliaSearch 5 client, we have the following type error:
```
Type '{ _ua: string; initAnalytics: (initOptions?: any) => AnalyticsClient; initPersonalization: (initOptions: any) => PersonalizationClient; initAbtesting: (initOptions?: any) => AbtestingClient; transporter: Transporter; appId: string; ... 67 more ...; updateApiKey({ key, apiKey }: UpdateApiKeyProps, requestOptions?: Re...' is not assignable to type 'SearchClient'.
  Types of property 'searchForFacetValues' are incompatible.
    Type '({ indexName, facetName, searchForFacetValuesRequest, }: SearchForFacetValuesProps, requestOptions?: RequestOptions | undefined) => Promise<SearchForFacetValuesResponse>' is not assignable to type 'undefined'.ts(2322)
InstantSearch.d.ts(43, 5): The expected type comes from property 'searchClient' which is declared here on type 'IntrinsicAttributes & UseInstantSearchApiProps<UiState, UiState> & { children?: ReactNode; }'
```

Functionally speaking, it seems like everything is working as expected but I might circle back as we implement more and more of Algolia's features.

**Reply by Haroenv** [reply]
Hm yes that seems like something wrong in the types, normally it should have inferred that you are using v5 and thus allow a client with only the `search` method. Do you have this reproducible somewhere? Otherwise I'm sure that if you silence the error, everything will work fine.

**Reply by aliou** [reply]
Hey @Haroenv, continuing the investigation, it seems like rendering `InstantSearch` anywhere in our app triggers the following type error:
```
Type '{ children: Element; as?: As | undefined; id?: string | undefined; is?: string | undefined; role?: AriaRole | undefined; title?: string | undefined; slot?: string | undefined; ... 676 more ...; _vertical?: SystemStyleObject | undefined; }' is not assignable to type 'IntrinsicAttributes & MergeWithAs<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>, any, Omit<...>, "symbol" | ... 177 more ...| "view">'.
  Type '{ children: Element; as?: As | undefined; id?: string | undefined; is?: string | undefined; role?: AriaRole | undefined; title?: string | undefined; slot?: string | undefined; ... 676 more ...; _vertical?: SystemStyleObject | undef
ined; }' is not assignable to type 'IntrinsicAttributes'.
    Types of property 'key' are incompatible.
      Type 'Key | null | undefined' is not assignable to type 'string | number | null | undefined'.
```

I managed to trace it back to this line in `@algolia/ui-components-shared`:
https://github.com/algolia/ui-components/blob/2cbcee8aa4207d6cabdf8162056437f12378e83d/packages/shared/src/Renderer.ts#L12

Funnily enough, it seems like you fixed the issue yesterday with the release 1.2.2 of the package. For now, we're going to try with the new version of `@algolia/ui-components-shared` for the `key` issue and ignore the type error for the `searchClient` prop.

I'll keep you up to date in the next following days on if this resolves the issue.

**Reply by Haroenv** [reply]
@aliou, yes that's just fixed, but not yet released in InstantSearch, will be soon!

**Reply by aliou** [reply]
@Haroenv Thanks for the fix! 

Regarding the initial issue of the type mismatch when passing the `algoliasearch` v5 client to the `InstantSearch` component. I've created a repo that reproduces the issue : https://github.com/aliou/algolia-instantsearch-type-issue-repro.

Let me know if you need more info!

**Reply by Haroenv** [reply]
Not sure yet what's not working related to `searchForFacetValues`, but I do know that `liteClient` works, which is generally recommended because it's much smaller:

```js
import { InstantSearch } from 'react-instantsearch-core';
import { liteClient } from 'algoliasearch/lite';

const searchClient = liteClient('', '');

function App() {
  return (
    <InstantSearch searchClient={searchClient} indexName="">
      <div>hello</div>
    </InstantSearch>
  );
}
```

**Reply by aliou** [reply]
Hey @Haroenv, we're investigating moving to another provider, so you can close this discussion thread. Thanks!

**Reply by Haroenv** [reply]
Discussions can't be closed, but I hope you're not looking to move providers just because of this issue?
